README.md

```diff
@@ -107,6 +107,35 @@ end
 
 Iteration hooks into Sidekiq and Resque out of the box to support graceful interruption. No extra configuration is required.
 
+## Stopping the iteration
+
+Sometimes we want to stop the execution of our job, Iteration use `throw` and `catch` to terminate the execution early.
+
+At any moment while iterating you use `throw(:abort)` the execution will be terminated. The complete callback will be executed as if the job has been completed.
+
+```ruby
+class BatchesJob < ActiveJob::Base
+  include JobIteration::Iteration
+
+  def build_enumerator(cursor:)
+    enumerator_builder.active_record_on_batches(
+      Product.all,
+      cursor: cursor,
+      batch_size: 3
+    )
+  end
+
+  def each_iteration(record)
+    self.class.records_performed << record
+    throw(:abort) if self.class.records_performed.size == 2
+  end
+end
+```
+
+If you want to terminate the execution of the job and skip the complete callback use `throw(:abort, :skip_complete_callbacks)`
+
+Using `throw` and `catch` instead of Exceptions is a deliberated decision, by design using exceptions as control flow is a bad design pattern, also the performance implications of using exceptions on a library like this one are important. Every time we raise an Exception ruby has to build a stack trace, so avoiding it will help us.
```

Comment:
I'm not sure if we need this clarification in the README. It would become quite large if we documented every design decision there.

Reply:
Since it's not a great design and we're not sure if we'll have those callbacks in future releases of gem (https://github.com/Shopify/job-iteration/issues/19), I'd avoid promoting it to users.

Reply:
Fantastic. Let's remove it.